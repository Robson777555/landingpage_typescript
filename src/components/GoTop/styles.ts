import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;