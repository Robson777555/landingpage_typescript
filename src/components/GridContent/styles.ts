// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;