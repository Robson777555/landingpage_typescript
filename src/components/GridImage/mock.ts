interface GridItem {
  altText: string;
  srcImg: string;
}

interface MockData {
  title: string;
  description: string;
  grid: GridItem[];
  background?: boolean;
}

const mock: MockData = {
  background: false,
  title: '<b style="font-size: 2rem;">Tec</b>',
  description: 'Conhe√ßa as principais tecnologias usadas neste projeto:',
  grid: [
    {
      altText: 'JavaScript Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/6/6a/JavaScript-logo.png',
    },
    {
      altText: 'React Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg',
    },
    {
      altText: 'Next.js Logo',
      srcImg: 'https://assets.vercel.com/image/upload/v1662090959/front/nextjs/twitter-card.png',
    },
    {
      altText: 'Node.js Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/d/d9/Node.js_logo.svg',
    },
    {
      altText: 'Git Logo',
      srcImg: 'https://git-scm.com/images/logos/downloads/Git-Icon-1788C.png',
    },
  ],
};

export default mock;