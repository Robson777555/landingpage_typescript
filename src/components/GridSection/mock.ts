interface GridItem {
  title: string;
  description: string;
}

interface MockData {
  title: string;
  description: string;
  grid: GridItem[];
  background?: boolean;
}

const mock: MockData = {
  title: '<b style="font-size: 2rem;">Serviço</b>',
  description: 'Conheça as atividades podem ser feitas para melhorar o seu negócio.',
  background: false,
  grid: [
    {
      title: '<b style="font-size: 1.2rem;">Web</b>',
      description: 'Sites e aplicações web personalizadas com as melhores tecnologias do mercado, garantindo performance e segurança para seu negócio.'
    },
    {
      title: '<b style="font-size: 1.2rem;">Análise</b>',
      description: 'Análise especializada para identificar as melhores soluções tecnológicas para suas necessidades específicas.'
    },
    {
      title: '<b style="font-size: 1.2rem;">Otimiza</b>',
      description: 'Melhoramos a performance e experiência do usuário em suas plataformas existentes.'
    },
  ],
};

export default mock;