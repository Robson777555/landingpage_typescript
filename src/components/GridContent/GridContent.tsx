'use client';
import React from 'react';
import * as Styled from './styles';
import { SectionBackground } from '../SectionBackground/SectionBackground';
import { Heading } from '../Heading/Heading';
import { TextComponent } from '../TextComponent/TextComponent';

interface GridContentProps {
  title: string;
  html: string;
  background?: boolean;
}

export const GridContent = ({ 
  title, 
  html, 
  background = false 
}: GridContentProps) => {
  return (
    <SectionBackground background={background}>
      <Styled.Container>
        <Heading uppercase colorDark={!background} as="h2">
          {title}
        </Heading>
        <Styled.HtmlContainer>
          <TextComponent>{html}</TextComponent>
        </Styled.HtmlContainer>
      </Styled.Container>
    </SectionBackground>
  );
};