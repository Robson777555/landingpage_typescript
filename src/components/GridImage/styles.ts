// src/components/GridImage/styles.js
import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.large};
      justify-items: center;
    }
  `}
`;

export const GridElement = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    border-radius: ${theme.spacings.small};
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
    transition: transform 300ms ease-in-out;
    width: 100%;
    max-width: 340px;
    margin: 0 auto;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-5px);
    }

    img, .next-image {
      max-width: 220px;
      width: 100%;
      height: auto;
      margin: 0 auto;
      display: block;
    }

    h2, h3, h4, h5, h6 {
      font-size: 2.4rem;
      font-weight: bold;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }

    @media (max-width: 768px) {
      padding: ${theme.spacings.small};
      max-width: 98vw;
      min-width: 0;
      h2, h3, h4, h5, h6 {
        font-size: 2rem;
        font-weight: bold;
        text-align: center;
        white-space: normal;
        word-break: break-word;
        width: 100%;
        line-height: 1.2;
      }
    }
  `}
`;
