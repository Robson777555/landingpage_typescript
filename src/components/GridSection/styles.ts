import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.xlarge};
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    padding: ${theme.spacings.medium};
    background: ${theme.colors.white};
    border-radius: ${theme.spacings.small};
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
    overflow: hidden;

    &:hover {
      transform: translateY(-5px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.15);
    }

    ${HeadingContainer} {
      position: relative;
      left: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: 2.4rem;
      text-align: center;
      word-break: break-word;
      white-space: normal;
      font-weight: bold;
      @media (max-width: 768px) {
        font-size: 2rem;
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4rem;
      top: -2rem;
      left: 0.5rem;
      transform: rotate(0deg);
      color: ${theme.colors.mediumGray};
      opacity: 0.3;
      z-index: 0;
      @media (max-width: 768px) {
        font-size: 2.5rem;
        top: -1.2rem;
        left: 0.5rem;
      }
    }

    ${TextComponent} {
      text-align: left;
      font-size: 1.2rem;
      @media (max-width: 768px) {
        font-size: 1rem;
      }
    }
  `}
`;

export const Title = styled.h3`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
