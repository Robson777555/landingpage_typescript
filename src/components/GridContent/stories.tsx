import React from 'react';
import { GridContent } from './GridContent';
import mock from './mock';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof GridContent> = {
  title: 'GridContent',
  component: GridContent,
  args: mock,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

type Story = StoryObj<typeof GridContent>;

export const Default: Story = {
  render: (args) => <GridContent {...args} />,
};

export const WithBackground: Story = {
  args: {
    ...mock,
    background: true,
  },
};