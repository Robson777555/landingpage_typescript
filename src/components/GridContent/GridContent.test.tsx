import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { GridContent } from './GridContent';
import { describe, it, expect } from 'vitest';
import mock from './mock';

describe('<GridContent />', () => {
  it('should render grid content with title and HTML', () => {
    render(
      <ThemeProvider theme={theme}>
        <GridContent {...mock} />
      </ThemeProvider>
    );

    expect(screen.getByRole('heading', { name: mock.title })).toBeInTheDocument();
    expect(screen.getByText(/Pellentesque habitant morbi tristique/i)).toBeInTheDocument();
  });

  it('should render without background when not provided', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GridContent {...mock} background={false} />
      </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyle({
      background: theme.colors.white
    });
  });

  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GridContent {...mock} />
      </ThemeProvider>
    );
    
    expect(container.firstChild).toMatchSnapshot();
  });
});