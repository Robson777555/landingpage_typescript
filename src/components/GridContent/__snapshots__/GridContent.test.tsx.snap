// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridContent /> > should match snapshot 1`] = `
<div
  class="sc-fmZSGO cqWQSY"
  data-testid="section-background"
>
  <div
    class="sc-gwsNht cxvrue"
    data-testid="section-container"
  >
    <div
      class="sc-bXdtCk eOffcb"
    >
      <h2
        class="sc-dYOLZc dwtOGd"
        data-testid="heading"
      >
        O título da Aplicação
      </h2>
      <div
        class="sc-hKDTPf eyKtpL"
      >
        <div
          class="sc-jMpmlX fCkSEf"
          data-testid="text-component"
        >
          <p>
            <strong>
              Pellentesque habitant morbi tristique
            </strong>
             senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. 
            <em>
              Aenean ultricies mi vitae est.
            </em>
             Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, 
            <code>
              commodo vitae
            </code>
            , ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. 
            <a
              href="#"
            >
              Donec non enim
            </a>
             in turpis pulvinar facilisis. Ut felis.
          </p>
          

  
          <h2>
            Header Level 2
          </h2>
          

  
          <ol>
            
    
            <li>
              Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
            </li>
            
    
            <li>
              Aliquam tincidunt mauris eu risus.
            </li>
            
  
          </ol>
          

  
          <blockquote>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
          </blockquote>
          

  
          <h3>
            Header Level 3
          </h3>
          

  
          <ul>
            
    
            <li>
              Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
            </li>
            
    
            <li>
              Aliquam tincidunt mauris eu risus.
            </li>
            
  
          </ul>
          

  
          <pre>
            <code>
              const example = "This is a code block";
            </code>
          </pre>
        </div>
      </div>
    </div>
  </div>
</div>
`;
